common/persistence/elasticsearch/esVisibilityStore.go

```diff
@@ -116,6 +119,7 @@ func (v *esVisibilityStore) RecordWorkflowExecutionStarted(
 	request *p.InternalRecordWorkflowExecutionStartedRequest,
 ) error {
 	v.checkProducer()
+	memo := v.serializeMemo(thrift.FromMemo(request.Memo), request.DomainUUID, request.WorkflowID, request.RunID)
```

Comment:
Looks to me that everywhere you call serializeMemo you call thrift.FromMemo - probably it makes sense to just pass request.Memo and call thrift.FromMemo in serializeMemo